source/extensions/filters/http/composite/filter.cc

```diff
@@ -95,6 +95,58 @@ void Filter::encodeComplete() {
 void Filter::onMatchCallback(const Matcher::Action& action) {
   const auto& composite_action = action.getTyped<ExecuteFilterAction>();
 
+  // Handle named filter chain lookup.
+  if (composite_action.isNamedFilterChainLookup()) {
+    // Check sampling first.
+    if (composite_action.actionSkip()) {
+      return;
+    }
+
+    const std::string& chain_name = composite_action.filterChainName();
+
+    // Soft fail: if no named filter chains are configured, do nothing.
+    if (!named_filter_chains_) {
+      ENVOY_LOG(debug, "filter_chain_name '{}' specified but no named filter chains configured",
+                chain_name);
+      return;
+    }
+
+    // Look up the filter chain by name.
+    auto it = named_filter_chains_->find(chain_name);
+    if (it == named_filter_chains_->end()) {
+      // Soft fail: if the named filter chain is not found, do nothing.
+      ENVOY_LOG(debug, "filter_chain_name '{}' not found in named filter chains", chain_name);
+      return;
+    }
+
+    // Create filters from the pre-compiled factories.
+    FactoryCallbacksWrapper wrapper(*this, dispatcher_, true /* is_filter_chain */);
+    for (const auto& factory_cb : it->second) {
+      factory_cb(wrapper);
+    }
+
+    if (!wrapper.errors_.empty()) {
+      stats_.filter_delegation_error_.inc();
+      ENVOY_LOG(debug, "failed to create filters from named filter chain '{}': {}", chain_name,
+                accumulateToString<absl::Status>(
+                    wrapper.errors_, [](const auto& status) { return status.ToString(); }));
+      return;
+    }
```

Comment:
When will this happen? I initially thought there should no any error at runtime because all errors should be processed when the configuration was loading?